code/datums/spells/infinite_guns.dm

```diff
@@ -17,5 +19,41 @@
 		C.drop_item()
 		C.swap_hand()
 		C.drop_item()
-		var/obj/item/gun/projectile/shotgun/boltaction/enchanted/GUN = new
+		var/obj/item/gun/projectile/shotgun/boltaction/enchanted/GUN = new gun_type()
 		C.put_in_hands(GUN)
+
+/obj/effect/proc_holder/spell/infinite_guns/fireball
+	name = "Rapid-fire Fireball"
+	desc = "Multiple Fireballs. Need I explain more? Requires both hands free to use."
+	invocation_type = "none"
+
+	school = "evocation"
```

Comment:
```suggestion
```
unused and redeclaring

Reply:
schools are cool, I'll fix the invocation_type though.